dbLayer/src/main/java/com/plhal/ares/dblayer/PredmetPodnikani.java

```diff
@@ -18,6 +19,7 @@
     private String nazev;
 
     @ManyToMany(mappedBy = "predmetPodnikani")
+    @JsonBackReference
```

Comment:
Oddelte prosim od sebe json objekty a databazove objekty. Tento pristup je problemovy z par pohledu:
1. nechtena zavislost v db modulu (json)
2. Ze servisni vrstvy musite zverejnovat datove objekty, ktere umoznuji praci s databazi.
3. Nutnost na vsech vrstvach (i WS) pracovat s objekty z nejnizsi vrstvy.

Resenim je v podstate existence "stejnych" objekty (jedny jsou databazove a druhe servisni). Na servisni si vytvorit instanci a prekopirovat tam ty datove a predat je to WS. pro jednoduchost muzeme dat json anotace do tech servisnich, pro uplne oddeleni bychom tam museli mit ty objekty trikrat (databazove, servisni a ty ve WS anotovane json anotacema).

Reply:
To znamená, že v servisním modulu vytvořím nové třídy (firma, predmetpodnikani a statutarniorgan). Do nich nakopíruji hodnoty z databázových objektů a tyto nové objekty potom předám dále do rest modulu?

Reply:
Presne. Nemusite se obavat, ze je to "kopie" (coz v tendle moment bude), protoze u vetsich aplikaci si nemusi servisni a datova vrstva odpovidat. Muze mit nejake veci navic a tak podobne. Napriklad ve firme neni uplne dobry napad pouzivat jako primarni klic ico (nicmene neni to treba neak resit) - spis si zavest databazove id. U aplikace tohoto typu (databaze je jen "cache") byste pak na servisni vrstve to databazove id vubec nepotreboval.